all: TestFindLatestImage fails at HEAD when running go test ./...

`TestFindLatestImage` fails when running `go test ./...` on a fresh clone of https://github.com/googleapis/librarian with the following errors:

```
--- FAIL: TestFindLatestImage (0.00s)
    --- FAIL: TestFindLatestImage/AR_pinned (0.34s)
        system_test.go:493: FindLatestImage() error = rpc error: code = PermissionDenied desc = Permission 'artifactregistry.versions.list' denied on resource '//artifactregistry.googleapis.com/projects/cloud-sdk-librarian-prod/locations/us-central1/repositories/images-prod/packages/librarian-go' (or it may not exist).
            error details: name = ErrorInfo reason = IAM_PERMISSION_DENIED domain = artifactregistry.googleapis.com metadata = map[permission:artifactregistry.versions.list resource:projects/cloud-sdk-librarian-prod/locations/us-central1/repositories/images-prod/packages/librarian-go]
    --- FAIL: TestFindLatestImage/AR_unpinned (0.38s)
        system_test.go:493: FindLatestImage() error = rpc error: code = PermissionDenied desc = Permission 'artifactregistry.versions.list' denied on resource '//artifactregistry.googleapis.com/projects/cloud-sdk-librarian-prod/locations/us-central1/repositories/images-prod/packages/librarian-go' (or it may not exist).
            error details: name = ErrorInfo reason = IAM_PERMISSION_DENIED domain = artifactregistry.googleapis.com metadata = map[permission:artifactregistry.versions.list resource:projects/cloud-sdk-librarian-prod/locations/us-central1/repositories/images-prod/packages/librarian-go]
2025/10/29 23:06:28 INFO Checking for repository dir=/var/folders/_j/n7nk6w3x6t3f0xjc0nz2chr80000gn/T/TestGitCheckoutknown_SHA260199818/001/librarian
2025/10/29 23:06:28 INFO Checking for repository dir=/var/folders/_j/n7nk6w3x6t3f0xjc0nz2chr80000gn/T/TestGitCheckoutunknown_SHA3275039683/001/librarian
2025/10/29 23:06:28 INFO Repository not found, executing clone
...
```

--- a/system_test.go
+++ b/system_test.go
@@ -448,7 +448,7 @@ func TestCreateRelease(t *testing.T) {
 }
 
 func TestFindLatestImage(t *testing.T) {
-	t.Skip("Temporarily skipping until we can run it reliably. See https://github.com/googleapis/librarian/issues/2720")
+	t.Logf("Note: This test requires authentication with the Artifact Registry in project 'cloud-sdk-librarian-prod'.")
 
 	// If we are able to configure system tests on GitHub actions, then update this
 	// guard clause.

